widget/view.controller.js

```diff
@@ -0,0 +1,146 @@
+          })
+          .catch(error => {
+            console.log(error);
+            installSequentially(index + 1);
+          });
+      }
+      installSequentially(0);
+    }
+
+    function backToStartPage() {
+      WizardHandler.wizard('timSolutionpackWizard').previous();
+    }
+
+    function backToinstallConnector() {
+      $scope.feedConnectors.sort((firstItem, secondItem) => secondItem.installed - firstItem.installed);
+      $scope.allConnectorsInstalled = false;
+      WizardHandler.wizard('timSolutionpackWizard').previous();
+    }
+
+    function moveToSelectConnector() {
+      threatIntelManagementConfigurationService.getFeedConnectors('Fortinet FortiGuard Threat Intelligence').then(function (response) {
+        $scope.fortiGuardConector = response.data['hydra:member'][0];
+        if ($scope.fortiGuardConector.installed === false) {
+          $scope.isFortiGuardConectorInsttalled = false;
+          threatIntelManagementConfigurationService.installConnector($scope.fortiGuardConector)
+            .then(resp => {
+              const importJobId = resp.data.importJob;
+              threatIntelManagementConfigurationService.getInstallationProgress(importJobId, $scope.fortiGuardConector)
+                .then(() => {
+                  //$scope.fortiGuardConector.selectConnector = true;
+                  threatIntelManagementConfigurationService.getFeedConnectors().then(function (response) {
+                    $scope.feedConnectors = response.data['hydra:member'];
+                    $scope.feedConnectors.sort((firstItem, secondItem) => secondItem.installed - firstItem.installed);
+                    //$scope.installedConnectors = $scope.feedConnectors.filter(connector => connector.installed === true);
+                    $scope.feedConnectors.forEach(feedConnector => {
+                      //feedConnector.toggleSelectFeeds = false;
+                      $scope.isFortiGuardConectorInsttalled = true;
+                      feedConnector.selectConnector = feedConnector.installed ? true : false;
+                      WizardHandler.wizard('timSolutionpackWizard').next();
+                    });
+                  });
+                })
+                .catch(error => {
+                  console.log(error);
+                  // Continue to the next connector even if there is an error
+                });
+            })
+            .catch(error => {
+              console.log(error);
+              // Continue to the next connector even if there is an error
+            });
+        }
+        else {
+          threatIntelManagementConfigurationService.getFeedConnectors().then(function (response) {
+            $scope.feedConnectors = response.data['hydra:member'];
+            $scope.feedConnectors.sort((firstItem, secondItem) => secondItem.installed - firstItem.installed);
+            $scope.feedConnectors.forEach(feedConnector => {
+              $scope.isFortiGuardConectorInsttalled = true;
```

Comment:
this can be optimised further and avoid fetching feed connectors again and again and marking them as installe or not installed.